lib/src/ruby/module.cc

```diff
@@ -170,8 +173,16 @@ namespace facter { namespace ruby {
             throw runtime_error("Ruby API is not initialized.");
         }
 
-        // Initialize the search paths
-        initialize_search_paths(paths);
+        // Load global settigs from config file
+        load_global_settings(load_default_config_file(), _config_file_settings);
+
+        // Initialize custom fact paths
+        vector<string> custom_fact_path = paths;
+        if (_config_file_settings.count("custom-dir")) {
+            auto config_paths = _config_file_settings["custom-dir"].as<vector<string>>();
+            custom_fact_path.insert(custom_fact_path.end(), config_paths.begin(), config_paths.end());
```

Comment:
Can you verify if `vector::insert` does one allocation or N when passed a random access iterator pair?